File: extra samples/vb/TestDemoButtons/DemoButtonsForm.vb
'
' DemoButtonsForm.vb
'
' Allow user to input a signature
'
' Copyright (c) 2015 Wacom GmbH. All rights reserved.
'
'
Public Class DemoButtonsForm

    Private Sub DemoButtonsForm_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles MyBase.Load
        print("Ready")
    End Sub

    Private Sub btnSign_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnSign.Click
        Dim usbDevices = New wgssSTU.UsbDevices()
        If (usbDevices.Count <> 0) Then
            Try
                print("Found " + usbDevices.Count.ToString() + " devices")
                Dim usbDevice = usbDevices(0)
                Dim demo = New SignatureForm(Me, usbDevice)
                Dim res = demo.ShowDialog()
                print("SignatureForm returned: " + res.ToString())
                If (res = DialogResult.OK) Then
                    DisplaySignature(demo)
                End If

            Catch ex As Exception
                MsgBox("Exception: " + ex.Message)
            End Try

        Else
            print("No devices found")
        End If
    End Sub

    Private Sub btnClose_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnClose.Click
        Me.Close()
    End Sub
    Public Sub print(ByVal txt As String)
        txtDisplay.Text += txt + vbCrLf
        txtDisplay.SelectionStart = txtDisplay.Text.Length ' scroll to end
        txtDisplay.ScrollToCaret()
    End Sub
    Private Sub DisplaySignature(ByVal demo As SignatureForm)
        Dim bitmap As Bitmap

        bitmap = demo.GetSigImage()
        ' resize the image to fit the screen
        Dim scale = 2       ' halve or quarter the image size
        If bitmap.Width > 400 Then
            scale = 4
        End If
        PictureBox1.Size = New Size(bitmap.Width / scale, bitmap.Height / scale)
        PictureBox1.Image = bitmap
        'centre the image in the panel
        Dim x, y As Integer
        x = Panel1.Location.X + ((Panel1.Width - PictureBox1.Width) / 2)
        y = Panel1.Location.Y + ((Panel1.Height - PictureBox1.Height) / 2)
        Me.PictureBox1.Location = New Point(x, y)

        PictureBox1.BringToFront()

        'bitmap.Save("C:\\temp\\sig.png", Imaging.ImageFormat.Png) ' to save the image to disk
    End Sub
End Class

File: extra samples/csharp/TestDemoButtons/DemoButtonsForm.cs
/******************************************************* 

  DemoButtonsForm.cs
  
  Controlling function for C# Demobuttons program allowing
  user to capture a signature with 3 control buttons
  
  Copyright (c) 2023 Wacom Ltd. All rights reserved.
  
********************************************************/
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

enum PenDataOptionMode
{
  PenDataOptionMode_None,
  PenDataOptionMode_TimeCount,
  PenDataOptionMode_SequenceNumber,
  PenDataOptionMode_TimeCountSequence
};

namespace DemoButtons
{
    public partial class DemoButtonsForm : Form
    {
        public DemoButtonsForm()
        {
            string STU_ModelName;

            InitializeComponent();
            STU_ModelName = TabletLib.STU_Tablet.FindSTUModelName();
            txtSTUModel.Text = STU_ModelName;

            if (STU_ModelName == "STU-500" || STU_ModelName == "STU-300")
              chkCalcSpeed.Enabled = false;

            if (STU_ModelName == "STU-300")
              chkReverseImage.Enabled = false;
        }

        private void btnSign_Click(object sender, EventArgs e)
        {

            wgssSTU.UsbDevices usbDevices = new wgssSTU.UsbDevices();
            if (usbDevices.Count != 0)
            {
                try
                {
                    wgssSTU.IUsbDevice usbDevice = usbDevices[0]; // select a device

                    SignatureForm demo = new SignatureForm(this, usbDevice, chkSaveImage.Checked, chkCalcSpeed.Checked, chkReverseImage.Checked);
                    DialogResult res = demo.ShowDialog();
                    print("SignatureForm returned: " + res.ToString());
                    if (res == DialogResult.OK)
                    {
                        txtSTUModel.Text = demo.stu_Tablet.information.modelName;

                        if (demo.stu_Tablet.information.modelName == "STU-500" || demo.stu_Tablet.information.modelName == "STU-300")
                          chkCalcSpeed.Enabled = false;

                        if (demo.stu_Tablet.information.modelName == "STU-300")
                          chkReverseImage.Enabled = false;

                        DisplaySignature(demo);
                    }
                    demo.Dispose();
                }
                catch (Exception ex)
                {
                    MessageBox.Show(ex.Message);
                }
            }
            else
            {
                print("No STU devices attached");
            }
        }

        public void print(string txt)
        {
            txtDisplay.Text += txt + "\r\n";
            txtDisplay.SelectionStart = txtDisplay.Text.Length; // scroll to end
            txtDisplay.ScrollToCaret();
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            this.Close();
        }
        
        private void DisplaySignature(SignatureForm demo )
        {
          Bitmap bitmap;

          bitmap = GraphicsLib.GraphicFunctions.GetSigImage(this, demo, demo.stu_Tablet);
          // resize the image to fit the screen
          int scale = 2;       // halve or quarter the image size
          if( bitmap.Width > 400 )
              scale = 4;
          pictureBox1.Size = new Size(bitmap.Width / scale, bitmap.Height / scale);
          pictureBox1.SizeMode = PictureBoxSizeMode.StretchImage;
          pictureBox1.Image = bitmap;
          pictureBox1.Parent = this;
          //centre the image in the panel
          int x, y;
          x = panel1.Location.X + ((panel1.Width - pictureBox1.Width) / 2);
          y = panel1.Location.Y + ((panel1.Height - pictureBox1.Height) / 2);
          this.pictureBox1.Location = new Point(x, y);
          pictureBox1.BringToFront();
        }
    }
    
}

